lib/elixir/lib/enum.ex

```diff
@@ -1099,6 +1099,14 @@ defmodule Enum do
     collectable ++ to_list(enumerable)
   end
 
+  def into(enumerable, collectable) when is_binary(collectable) do
+    postfix =
+      enumerable
+      |> to_list
+      |> IO.iodata_to_binary
+    collectable <> postfix
```

Comment:
The iodata to binary conversion can be moved down here to possibly avoid one extra allocation `IO.iodata_to_binary([collectable | postfix])`.